frontend/src/scenes/billing/billingLogic.ts

```diff
@@ -127,28 +129,45 @@ export const billingLogic = kea<billingLogicType>([
             },
         ],
     }),
-    loaders(({ actions }) => ({
+    loaders(({ actions, values }) => ({
         billing: [
             null as BillingV2Type | null,
             {
                 loadBilling: async () => {
+                    actions.resetBillingError()
                     const response = await api.get('api/billing-v2')
 
                     return parseBillingResponse(response)
                 },
 
                 updateBillingLimits: async (limits: { [key: string]: string | null }) => {
+                    actions.resetBillingError()
                     const response = await api.update('api/billing-v2', { custom_limits_usd: limits })
 
                     lemonToast.success('Billing limits updated')
                     return parseBillingResponse(response)
                 },
 
                 deactivateProduct: async (key: string) => {
-                    const response = await api.get('api/billing-v2/deactivate?products=' + key)
-                    lemonToast.success('Product unsubscribed')
-                    actions.reportProductUnsubscribed(key)
-                    return parseBillingResponse(response)
+                    actions.resetBillingError()
+                    try {
+                        const response = await api.getResponse('api/billing-v2/deactivate?products=' + key)
+                        const jsonRes = await getJSONOrNull(response)
+                        lemonToast.success('Product unsubscribed')
+                        actions.reportProductUnsubscribed(key)
+                        return parseBillingResponse(jsonRes)
+                    } catch (error: any) {
+                        lemonToast.error(`${error.detail}`, {
+                            button: {
+                                label: 'See Invoices',
+                                action: () =>
+                                    (window.location.href = values.billing?.stripe_portal_url || window.location.href),
+                            },
+                        })
```

Comment:
I wonder if a toast is the best way to communicate errors here. They're kind of hard to catch sometimes. Maybe the general error at the top is better? Or even on the unsub modal under the cancel button. If the modal doesn't close, they know the cancel hasn't happened. Otherwise you might think "it closed, so it's complete" and ignore any errors that pop up.

Reply:
Ok, good points. I had debated also showing the error in an error banner at the top of the page but felt it was perhaps too obnoxious. I think showing it as part of the unsubscribe modal is probably the best solution.